lib/config/configitem.cpp

```diff
@@ -472,13 +472,16 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 				continue;
 
 			std::atomic<int> committed_items(0);
-			std::mutex newItemsMutex;
 
 			{
 				auto items (itemsByType.find(type.get()));
 
 				if (items != itemsByType.end()) {
-					upq.ParallelFor(items->second, [&committed_items, &newItems, &newItemsMutex](const ItemPair& ip) {
+					for (const ItemPair& pair: items->second) {
+						newItems.emplace_back(pair.first);
+					}
```

Comment:
OK, so originally we newItems-ified all, then (400117e2f63aaa3523b77acb627236ec8fb9880b) only committed ones. And now ☝️ all items which are **going** to be committed. This is necessary for the case the below `item->Commit()` throws and CommitNewItems() returns false, so that we remember to unregister them (again). So far so good.

Neither this is a regression, as f7298e85d2c072c5423dfb1e21d11bab953e3475 was the main part of #9406 and it wasn't reverted here. 👍